modules/icgc-argo-workflows/checkinput/resources/usr/bin/variantcallqc.py

```diff
@@ -0,0 +1,319 @@
+    """
+    Define a service that can validate and transform each given row.
+
+    Attributes:
+        modified (list): A list of dicts, where each dict corresponds to a previously
+            validated and transformed row. The order of rows is maintained.
+
+    """
+
+    VALID_FORMATS = (
+        "bcf",
+        "vcf",
+        "vcf.gz",
+        "vcf.csi",
+        "vcf.gz.csi",
+        "vcf.tbi",
+        "vcf.gz.tbi",
+        "bcf.tbi"
+        "bcf.csi"
+        ""
+    )
+
+    def __init__(
+        self,
+        analysis_type_col = 'analysis_type',
+        study_id_col = 'study_id',
+        patient_col = 'patient',
+        sex_col = 'sex',
+        status_col = 'status',
+        sample_col = 'sample',
+        experiment_col = 'experiment',
+        vcf_col = 'vcf',
+        vcf_index_col = 'vcf_index',
+        analysis_json_col = 'analysis_json',
+        **kwargs,
+    ):
+        """
+        Initialize the row checker with the expected column names.
+        analysis_type,study_id,patient,sex,status,sample,experiment,analysis_json
+        Args:
+            sample_col (str): The name of the column that contains the sample name
+                (default "sample").
+            first_col (str): The name of the column that contains the first (or only)
+                FASTQ file path (default "fastq_1").
+            second_col (str): The name of the column that contains the second (if any)
+                FASTQ file path (default "fastq_2").
+            single_col (str): The name of the new column that will be inserted and
+                records whether the sample contains single- or paired-end sequencing
+                reads (default "single_end").
+
+        """
+        super().__init__(**kwargs)
+        self._analysis_type_col = analysis_type_col
+        self._study_id_col = study_id_col
+        self._patient_col = patient_col
+        self._sex_col = sex_col
+        self._status_col = status_col
+        self._sample_col = sample_col
+        self._experiment_col = experiment_col
+        self._vcf_col = vcf_col
+        self._vcf_index_col = vcf_index_col
+        self._analysis_json_col = analysis_json_col
+        self._seen = []
+        self.modified = []
+
+
+    def validate_and_transform(self, row):
+        """
+        Perform all validations on the given row and insert the read pairing status.
+
+        Args:
+            row (dict): A mapping from column headers (keys) to elements of that row
+                (values).
+
+        """
+        self._validate_analysis_type(row) if row.get(self._analysis_type_col) else ""
+        self._validate_study_id(row) if row.get(self._study_id_col) else ""
+        self._validate_patient(row) if row.get(self._patient_col) else ""
+        self._validate_sex(row) if row.get(self._sex_col) else ""
+        self._validate_status(row) if row.get(self._status_col) else ""
+        self._validate_sample(row)
+        self._validate_experiment(row)
```

Comment:
`experiment` needs default when not provided.

Reply:
Similar issue for `vcf_index`. I got error when I ran with a samplesheet only contains columns: sample, vcf
```
KeyError: 'vcf_index'
```